kivy/uix/image.py

```diff
@@ -205,13 +255,23 @@ def get_image_ratio(self):
     def get_norm_image_size(self):
         if not self.texture:
             return list(self.size)
+
         ratio = self.image_ratio
         w, h = self.size
         tw, th = self.texture.size
 
+        if self.fit_mode == "cover":
+            widget_ratio = w / max(1, h)
+            if widget_ratio > ratio:
+                return [w, (w * th) / tw]
+            else:
+                return [(h * tw) / th, h]
+
         # ensure that the width is always maximized to the container width
-        if self.allow_stretch:
-            if not self.keep_ratio:
+        # NOTE: self.allow_stretch and self.keep_ratio conditions should be
+        # removed along with the deprecated properties in the future
+        if self.fit_mode in ("fill", "contain") or self.allow_stretch:
+            if self.fit_mode == "fill" or not self.keep_ratio:
```

Comment:
This will print the following during runtime:
```bash
[WARNING] Deprecated property "<BooleanProperty name=allow_stretch>" of object "<kivy.uix.image.Image object at 0x10d63cc90>" was accessed, it will be removed in a future version
```

Is better to avoid this behavior, and introduce a different migration method, which is accessed only if the developer specifies the `allow_stretch` or `keep_ratio` properties.

As an example, you can bind for changes of `allow_stretch` and `keep_ratio` and select the proper `fit_mode` so the `get_norm_image_size` is easier to read, and does not access `allow_stretch` and `keep_ratio`.